### PR TITLE
[snap] reduce pull load

### PR DESCRIPTION
**Pull request by Saviq**
Use `override-pull` to avoid pulling in any source when not needed.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -142,7 +142,6 @@ parts:
     - git
     - golang
     - libsystemd-dev
-    source: .
```
Comment (sergiusens) [inline snap/snapcraft.yaml]: I would leave `source: .` here to make this future proof (implicit sources are not a thing in snapcraft 3.0 when using bases)

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -156,6 +155,7 @@ parts:
 
   qemu:
     plugin: nil
+    override-pull: ""
```
Comment (sergiusens) [inline snap/snapcraft.yaml]: alternatively, create `snap/local` and use that as `source` and remove this stanza. But this implementation is future proof and one line removal away when moving to bases.

**Comment by Saviq** [comment]
bors r=townsend2010,sergiusens

**Comment by Saviq** [comment]
bors r=townsend2010,sergiusens

**Pull request merged**